### DISCUSSION TITLE
Skip kotlinFormat for KSP generated sources

### DISCUSSION THREAD

**Discussion by hantsy** [Kotlin/dataframe#348] [Q&A] [answered]
I have just upgraded DataFramework 0.10.0 to align with Kotlin 1.8.20, but I found former kolinter config does not work.

```
tasks.named<LintTask>("lintKotlinMain") {
    dependsOn("kspKotlin")
    source = (source - fileTree("$buildDir/generated")).asFileTree
}

tasks.named<FormatTask>("formatKotlinMain") {
    dependsOn("kspKotlin")
    source = (source - fileTree("$buildDir/generated")).asFileTree
}
```

When I ran `./gradlew clean formatKotlin`(I have to add `clean`, due to issue #349), there is a `> Task :formatKotlinGeneratedByKspKotlin` always to execute.

I ran the command `.gradlew tasks -all` and got two tasks in the result.

```bash
formatKotlinGeneratedByKspKotlin
formatKotlinGeneratedByKspTestKotlin
```

I tried to set up config for the task `formatKotlinGeneratedByKspKotlin`.

```
tasks.named<FormatTask>("formatKotlinGeneratedByKspKotlin") {
    dependsOn("kspKotlin")
    source = (source - fileTree("$buildDir/generated")).asFileTree
}
```

or 

```
tasks.withType<FormatTask>{
    dependsOn("kspKotlin")
    source = (source - fileTree("$buildDir/generated")).asFileTree
}
```

It does not work.

How to skip the Kotinter lint/format tasks for all generated files.

**Comment by Jolanrensen** [comment]
The linter tasks is controlled by `tasks.withType<LintTask> {}` and `kotlinter {}` in the :core build.gradle.kts. Currently we exclude generated files using the first. This excludes generated files from every LintTask instance, most probably including formatKotlinGeneratedByKspKotlin and formatKotlinGeneratedByKspTestKotlin.

**Comment by hantsy** [comment]
There is a new KSP plugin was just released. https://github.com/google/ksp/releases/tag/1.8.20-1.0.11 

I added the KSP plugin explicitly in my project.

```kotlin 
id("com.google.devtools.ksp") version "1.8.20-1.0.11"
```

But there is still no way to exclude the KSP generated files.

I tried to add the following suggested by https://github.com/jeremymailen/kotlinter-gradle/issues/322#issuecomment-1510745568

```kotlin 
tasks.named("formatKotlinGeneratedByKspKotlin").configure {
    enabled = false
}

tasks.named("formatKotlinGeneratedByKspTestKotlin").configure {
    enabled = false
}
```
And got the exception:

```bash
Task with name 'formatKotlinGeneratedByKspKotlin' not found in root project 'my-backend'.
````

**Comment by hantsy** [comment, marked as answer]
Finally add the following code fragment to skip the kotliner KSP related tasks.

```kotlin 
tasks.whenTaskAdded {
    // println("adding task: $name")
    if (name == "lintKotlinGeneratedByKspKotlin" ||
        name == "lintKotlinGeneratedByKspTestKotlin" ||
        name == "formatKotlinGeneratedByKspKotlin" ||
        name == "formatKotlinGeneratedByKspTestKotlin"
    ) {
        enabled = false
    }
}
```

**Reply by hantsy** [reply]
If these tasks are introduced by Dataframe plugin, I think Dataframe itself **should not** affect user's project settings.

**Comment by hantsy** [comment]
The `formatKotlinGeneratedByKspKotlin` and `formatKotlinGeneratedByKspTestKotlin` is introduced by DataFrame, or KSP plugin?

Is this issue fixed in the DataFrame 0.10.1?
